### PR TITLE
update optimmethod and metrics md for branch-0.1

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?
fix issue https://github.com/intel-analytics/BigDL/issues/1375
issue https://github.com/intel-analytics/BigDL/issues/1377
update optimmethod and metrics md for branch-0.1

### CODE REVIEW

File: docs/docs/APIdocs/Optimizers/Optim-Methods.md [hunk 1]
```diff
@@ -68,102 +50,66 @@ def rosenBrock(x: Tensor[Float]): (Float, Tensor[Float]) = {
...
 each iteration.
  
 **Scala example:**
 ```scala
-val optimMethod = new SGD[Float](learningRate= 1e-3,learningRateDecay=0.0,
-                               weightDecay=0.0,momentum=0.0,dampening=Double.MaxValue,
-                               nesterov=false,learningRateSchedule=Default(),
-                               learningRates=null,weightDecays=null)
-optimizer.setOptimMethod(optimMethod)
+optimizer.setOptimMethod(new SGD[Float]())
```
Comment (shane-huang) [inline docs/docs/APIdocs/Optimizers/Optim-Methods.md]: maybe we add a line about set state (learning rate, etc.)?

**Comment by shane-huang** [comment]
please also change ProgrammingGuide/optimization.md #resume-training
```
After training stops, you can resume from any saved point. Choose one of the model snapshots and the corresponding optim-method snapshot to resume (saved in checkpoint path, details see Checkpointing). Use Module.load (Scala) or Model.load(Python) to load the model snapshot into an model object, and OptimMethod.load (Scala only, Python does not support this yet) to load optimization method into an OptimMethod object. Then create a new Optimizer with the loaded model and optim method. Call Optimizer.optimize, and you will resume from the point where the snapshot is taken. Refer to OptimMethod Load and Model Load for details.

You can also resume training without loading the optim method, if you intend to change the learning rate schedule or even the optimization algorithm. Just create an Optimizer with loaded model and a new instance of OptimMethod (both Scala and Python).
```
